module/renderer/src/commonMain/kotlin/org/cru/godtools/shared/renderer/content/RenderSpacer.kt

```diff
@@ -0,0 +1,18 @@
+package org.cru.godtools.shared.renderer.content
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.ui.unit.dp
+import androidx.compose.foundation.layout.height
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import org.cru.godtools.shared.tool.parser.model.Spacer
+
+@Composable
+internal fun RenderSpacer(spacer: Spacer, modifier: Modifier = Modifier) {
+
+    when (spacer.mode) {
+        Spacer.Mode.FIXED -> Spacer(Modifier.height(spacer.height.dp))
+        Spacer.Mode.AUTO -> Column { Spacer(modifier = modifier.weight(1.0f)) }
```

Comment:
by creating another `Column` here you are localizing the `weight()` modifier to the local column. you really want to use the `ColumnScope` that comes from the outer `Column`. you can pass that into `RenderSpacer` by changing it to: `internal fun ColumnScope.RenderSpacer()`.

Reply:
Ah okay thanks I had a feeling that ColumnScope was needed since you had mentioned it in our call.  Going to read up on what you sent me.

Reply:
after you read up I can jump on a call if you have more specific questions.

internally the context receiver gets turned into a parameter sent to the function by the compiler, so `fun ColumnScope.RenderSpacer()` becomes `fun RenderSpacer($receiver: ColumnScope)`, and any calls that would use that context automagically just get called on that parameter

Reply:
Sweet ColumnScope did the trick:

<img width="1726" alt="column-scope-spacer" src="https://github.com/user-attachments/assets/5d62d3b8-b9ca-4fbf-a3f6-2856f71f7ded" />